crates/subspace-farmer/src/utils/farmer_piece_getter.rs

```diff
@@ -12,12 +12,16 @@ use subspace_networking::libp2p::kad::RecordKey;
 use subspace_networking::libp2p::PeerId;
 use subspace_networking::utils::multihash::ToMultihash;
 use subspace_networking::utils::piece_provider::{PieceProvider, PieceValidator, RetryPolicy};
+use subspace_networking::utils::random_walking_piece_provider::RandomWalkingPieceProvider;
 use subspace_networking::Node;
 use tracing::{debug, error, trace};
 
+const MAX_RANDOM_WALK_ROUNDS: usize = 50;
```

Comment:
Let's decrease this number to something like 5 or 10, 50 will be ridiculously slow, I can imagine how farmer would be better off reconstructing it at that point.